File: python/Reader/src/utils/ParseUtils.py
import sys
import os.path
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))

from model.SensorData import SensorData

class ParseUtils:

    @staticmethod
    def parseSensorData(bytes):
        temperature = int(bytes[2] + bytes[1], 16)
        if (temperature > 0x8000):
            temperature = temperature - 0x10000

        temperature = temperature / 10.0

        humidity = int(bytes[4], 16)
        battery = int(bytes[9], 16)
        return SensorData(temperature, humidity, battery)

File: c#/Reader/Reader/Util/ParseUtils.cs
using System.Diagnostics;
using Reader.Model;

namespace Reader.Util
{
    class ParseUtils
    {
        public static SensorData ParseSensorData(string[] bytes)
        {
            SensorData sensorData = new SensorData();

            var temperature = int.Parse(bytes[2] + bytes[1], System.Globalization.NumberStyles.HexNumber);
            if (temperature > 0x8000)
            {
                temperature = temperature - 0x10000;
            }
            sensorData.Temperature = temperature / 10.0;
            Debug.WriteLine("temperature: " + sensorData.Temperature);

            sensorData.Humidity = int.Parse(bytes[4] + "", System.Globalization.NumberStyles.HexNumber);
            Debug.WriteLine("humidity: " + sensorData.Humidity);

            sensorData.Battery = int.Parse(bytes[9] + "", System.Globalization.NumberStyles.HexNumber);
            Debug.WriteLine("battery: " + sensorData.Battery);

            return sensorData;
        }
    }
}
